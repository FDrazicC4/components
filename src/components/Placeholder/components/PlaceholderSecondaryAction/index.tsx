import React, { ReactElement } from 'react';
import clsx from 'clsx';
import Button from '../../../Button';
import { Props } from './types';


function PlaceholderPrimaryAction({
    className,
    onClick,
    children,
}: Props): ReactElement {


    const placeholderClasses = clsx(
        'brew-Placeholder__SecondaryAction',
        className,
    );


    return (
        <Button className={placeholderClasses} onClick={onClick}>{children}</Button>
    );


}


export default PlaceholderPrimaryAction;
