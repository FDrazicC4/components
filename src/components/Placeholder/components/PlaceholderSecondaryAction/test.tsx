import React from 'react';
import renderer from 'react-test-renderer';
import PlaceholderSecondaryAction from './index';
import { shallow } from 'enzyme';


describe('PlaceholderSecondaryAction', () => {

    it('renders correctly', () => {
        const tree = renderer
            .create(<PlaceholderSecondaryAction />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('mounts correctly', () => {
        const secondaryAction = shallow(<PlaceholderSecondaryAction />);
        expect(secondaryAction.hasClass('brew-Placeholder__SecondaryAction')).toBe(true);
    });

    it('correctly adds a className', () => {
        const className = `class-${Math.floor(Math.random() * 1000)}`;
        const secondaryAction = shallow(<PlaceholderSecondaryAction className={className} />);
        expect(secondaryAction.hasClass(className)).toBe(true);
    });

});
