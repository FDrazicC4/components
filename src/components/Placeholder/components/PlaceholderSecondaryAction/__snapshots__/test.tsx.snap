// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceholderSecondaryAction renders correctly 1`] = `
<div>
  <button
    className="Button Button__variant--standard Button__color--primary Button__size--medium brew-Placeholder__SecondaryAction"
    data-flip-config="{\\"translate\\":true,\\"scale\\":true,\\"opacity\\":true}"
    data-flip-id="wrapper"
    data-portal-key="portal"
    disabled={false}
    type="button"
  >
    <div
      className="Button__content"
      data-flip-config="{\\"translate\\":true,\\"scale\\":true,\\"opacity\\":true}"
      data-flip-id="content"
      data-portal-key="portal"
    >
      <span
        className="Button__text"
      >
        <span
          className="wrapper variant--inherit color--initial align--inherit display--initial"
        />
      </span>
    </div>
    <div
      className="Button__loadingIndicator"
      data-flip-config="{\\"translate\\":true,\\"scale\\":true,\\"opacity\\":true}"
      data-flip-id="loader"
      data-portal-key="portal"
    >
      <div
        className="brew-Progress--circular brew-Progress__color--primary brew-Progress__size--inherit"
      />
    </div>
  </button>
</div>
`;
